### PR TITLE
[Held requests] Hold Request education modal reappears after relogin

### PR DESCRIPTION
**Pull request by cdOut**
### Details

Save the action when the user dismisses the hold request education modal onto the backend, so we don't show it on every login interaction.

### Fixed Issues

$ https://github.com/Expensify/App/issues/37012
PROPOSAL:

### Tests

1. Request money from one account to another and then proceed to Hold it by going into the request and pressing the `Hold` button from the three dot menu.
2. After it displays make sure to either press the `Got it` button or dismiss the modal by pressing on the overlay.
3. Log out and then proceed to log in to the same account you just used
4. Navigate to any held request report and make sure the interstitial doesn't show up.

- [x] Verify that no errors appear in the JS console

### QA Steps

- [x] Verify that no errors appear in the JS console

### PR Author Checklist

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android: Native
    - [x] Android: mWeb Chrome
    - [x] iOS: Native
    - [x] iOS: mWeb Safari
    - [x] MacOS: Chrome / Safari
    - [x] MacOS: Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is either coming verbatim from figma or has been approved by marketing (in order to get marketing approval, ask the Bug Zero team member to add the Waiting for copy label to the issue)
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG)`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If the PR modifies the UI (e.g. new buttons, new UI components, changing the padding/spacing/sizing, moving components, etc) or modifies the form input styles:
    - [x] I verified that all the inputs inside a form are aligned with each other.
    - [x] I added `Design` label and/or tagged `@Expensify/design` so the design team can review the changes.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>

https://github.com/Expensify/App/assets/88325488/107ee460-ab64-4b32-b20b-443c9bd33bee

</details>

<details>
<summary>Android: mWeb Chrome</summary>

https://github.com/Expensify/App/assets/88325488/731c0806-6fba-43f2-8220-c19ec1d6a943

</details>

<details>
<summary>iOS: Native</summary>

https://github.com/Expensify/App/assets/88325488/4ab1f5d9-8faa-414f-a695-cb8cb4a53aa4

</details>

<details>
<summary>iOS: mWeb Safari</summary>

https://github.com/Expensify/App/assets/88325488/69ddc20f-36fd-470c-8658-ba4f942d19b6

</details>

<details>
<summary>MacOS: Chrome / Safari</summary>

https://github.com/Expensify/App/assets/88325488/80423017-0c42-4041-abca-e76970e8a04c

</details>

<details>
<summary>MacOS: Desktop</summary>

https://github.com/Expensify/App/assets/88325488/cba9eea5-7501-43c5-8089-87de99066966

</details>

### CODE REVIEW

**Review by BartoszGrajdek — APPROVED** [review]

**Comment by ahmedGaber93** [comment]
## Reviewer Checklist

- [ ] I have verified the author checklist is complete (all boxes are checked off).
- [ ] I verified the correct issue is linked in the `### Fixed Issues` section above
- [ ] I verified testing steps are clear and they cover the changes made in this PR
    - [ ] I verified the steps for local testing are in the `Tests` section
    - [ ] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [ ] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [ ] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [ ] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I verified tests pass on **all platforms** & I tested again on:
    - [ ] Android: Native
    - [ ] Android: mWeb Chrome
    - [ ] iOS: Native
    - [ ] iOS: mWeb Safari
    - [ ] MacOS: Chrome / Safari
    - [ ] MacOS: Desktop
- [ ] If there are any errors in the console that are unrelated to this PR, I either fixed them (preferred) or linked to where I reported them in Slack
- [ ] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [ ] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [ ] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [ ] I verified that comments were added to code that is not self explanatory
    - [ ] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [ ] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
    - [ ] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [ ] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is either coming verbatim from figma or has been approved by marketing (in order to get marketing approval, ask the Bug Zero team member to add the Waiting for copy label to the issue)
    - [ ] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [ ] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [ ] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [ ] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [ ] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [ ] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [ ] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [ ] If a new component is created I verified that:
    - [ ] A similar component doesn't exist in the codebase
    - [ ] All props are defined accurately and each prop has a `/** comment above it */`
    - [ ] The file is named correctly
    - [ ] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [ ] The only data being stored in the state is data necessary for rendering and nothing else
    - [ ] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [ ] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [ ] All JSX used for rendering exists in the render method
    - [ ] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [ ] If any new file was added I verified that:
    - [ ] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [ ] If a new CSS style is added I verified that:
    - [ ] A similar style doesn't already exist
    - [ ] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG`)
- [ ] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [ ] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [ ] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [ ] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [ ] If the PR modifies the UI (e.g. new buttons, new UI components, changing the padding/spacing/sizing, moving components, etc) or modifies the form input styles:
    - [ ] I verified that all the inputs inside a form are aligned with each other.
    - [ ] I added `Design` label and/or tagged `@Expensify/design` so the design team can review the changes.
- [ ] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [ ] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [ ] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>



</details>

<details>
<summary>Android: mWeb Chrome</summary>



</details>

<details>
<summary>iOS: Native</summary>



</details>

<details>
<summary>iOS: mWeb Safari</summary>



</details>

<details>
<summary>MacOS: Chrome / Safari</summary>



</details>

<details>
<summary>MacOS: Desktop</summary>



</details>

File: src/pages/ProcessMoneyRequestHoldPage.tsx [hunk 1]
```diff
@@ -15,7 +15,7 @@ function ProcessMoneyRequestHoldPage() {
     const {translate} = useLocalize();
 
     const onConfirm = useCallback(() => {
-        IOU.setShownHoldUseExplanation();
+        IOU.dismissHoldUseExplanation();
         Navigation.goBack();
     }, []);
```
Comment (ahmedGaber93) [inline src/pages/ProcessMoneyRequestHoldPage.tsx:21]: In wide screen we handle `onConfirm` here, but I think we may miss handling on modal dismiss

**Comment by ahmedGaber93** [comment]
I think we may have a race condition here.
1. hold request
2. after educational model appear, click Got it.
3. go back and reopen the hold request (modal not reopen)
4. go back and refresh the browser, then reopen the hold request (modal reappear reopen)

When I log `dismissedHoldUseExplanation` it seem it come with null for first times then back with the correct value.

https://github.com/Expensify/App/assets/41129870/3c81a0d0-d9b0-45d8-aad0-8e4586896c03

File: src/pages/ProcessMoneyRequestHoldPage.tsx (hunk 1 above)
Comment (cdOut) [reply src/pages/ProcessMoneyRequestHoldPage.tsx:21]: It is handled over here and then passed to `OnConfirm` and `OnClose` of the modal:

Screenshot 2024-06-06 at 10 16 11

**Comment by cdOut** [comment]
@robertjchen Could you please confirm the naming of the NVP and if it's currently in use on the main backend or is it up on staging? Is there a specific write command I'm supposed to use or is the one used for other NVPs okay?

File: src/pages/ProcessMoneyRequestHoldPage.tsx (hunk 1 above)
Comment (ahmedGaber93) [reply src/pages/ProcessMoneyRequestHoldPage.tsx:21]: @cdOut I think the code you mentioned is on small screen only, but wide screen only handle `onConfirm`, and not handle `OnClose`

 Screenshot 2024-06-06 at 8 15 02 PM

**Comment by robertjchen** [comment]
The naming looks correct to me 🤔  There's no specific write command, it should just be like the other NVPs.

**Comment by cdOut** [comment]
> When I log `dismissedHoldUseExplanation` it seem it come with null for first times then back with the correct value.

It's a pretty specific case that isn't easily replicable but I've added an initial value and that should fix the null value.

> After relogin, I am still can reproduce the issue, it seems backend not back `nvp_dismissedHoldUseExplanation` to the user. Did I miss something?

It seems to work for some of my accounts and not for the others, @robertjchen is there a specific behavior of how we handle NVPs on the backend here? For example when dealing with new accounts and old ones?

I'll investigate what exactly happens in this pipeline of reading the NVP from the backend to deciding whether to display the explanation modal.

File: src/pages/ProcessMoneyRequestHoldPage.tsx (hunk 1 above)
Comment (cdOut) [reply src/pages/ProcessMoneyRequestHoldPage.tsx:21]: In the screenshot you provided you can see an override for both the `onClose` and `onConfirm`, I am a little confused about what you exactly mean here? In my opinion all the closing / confirming conditions for both the widescreen and the mobile page / modal are met.

Comment (ahmedGaber93) [reply src/pages/ProcessMoneyRequestHoldPage.tsx:21]: @cdOut The code in my screenshot is for small screen, see line 263 on my screenshot `{isSmallScreenWidth && `

Comment (cdOut) [reply src/pages/ProcessMoneyRequestHoldPage.tsx:21]: Ah you're correct, sorry for the confusion! I'll add a handle for this case too.

Comment (ahmedGaber93) [reply src/pages/ProcessMoneyRequestHoldPage.tsx:21]: We handle the education modal in different places for wide screen we use `ProcessMoneyRequestHoldPage`, and for small screen we use `ProcessMoneyRequestHoldMenu`

Comment (cdOut) [reply src/pages/ProcessMoneyRequestHoldPage.tsx:21]: I am aware, I wrote the Educational Intersitial for the hold feature 😅. I’ll include this case in the full screen page.

**Comment by robertjchen** [comment]
> is there a specific behavior of how we handle NVPs on the backend here? For example when dealing with new accounts and old ones?

Hm, the NVP controls the display of the modal such that it shows up once for the lifetime of an account. Once the modal is dismissed, we set the NVP so that later on the UI we know not to ever show it again.

- For a new account, the NVP does not exist. In the frontend logic- we detect this and show the modal.
- For a pre-existing account, the NVP does not exist. In the frontend logic- we detect this and show the modal.

When the modal is dismissed we simply set the NVP to some true value. This is automatically propagated to the backend and saved.

Next time they log in, the NVPs are loaded into the frontend. The frontend detects that this NVP exists, and therefore does NOT show the modal.

Let me know if the helps clear it up 🙏  The NVP does not get any special treatment, it's just a setting we set once and check later on.

**Comment by ahmedGaber93** [comment]
> Next time they log in, the NVPs are loaded into the frontend. The frontend detects that this NVP exists, and therefore does NOT show the modal.

@robertjchen Could you please check why backend not back `nvp_dismissedHoldUseExplanation` for my account a.gaber.dev+1001@gmail.com?

- API request `api/SetNameValuePair` send the parameter `nvp_dismissedHoldUseExplanation` with true value and done successfully.

- The above request respond with empty `onyxData` `{"oldValue":"","jsonCode":200,"requestID":"89212ea2df3b536f-CAI","onyxData":[]}`
- After relogin `nvp_dismissedHoldUseExplanation` not back in `api/OpenApp`, I also checked browser OnyxDB and it not found.

**Comment by robertjchen** [comment]
Hm, could you try dropping the `nvp_` prefix and just using`dismissedHoldUseExplanation` to see if it comes back? 🙏

**Comment by ahmedGaber93** [comment]
@robertjchen `dismissedHoldUseExplanation` also not back in `api/OpenApp`, and not found in browser OnyxDB.

**Comment by robertjchen** [comment]
The team expert who designed the new NVP system just tested this PR as-is and said the NVP was being returned properly. Can you please test again and check if `nvp_dismissedHoldUseExplanation` is present in the OnyxDB? 😕

**Comment by ahmedGaber93** [comment]
I am in break now, and I will recheck after one hour.

**Comment by ahmedGaber93** [comment]
> The team expert who designed the new NVP system just tested the backend API and said the NVP was being returned properly for api/OpenApp. Can you please test again and check if nvp_dismissedHoldUseExplanation is present in the OnyxDB? 😕

@robertjchen `api/OpenApp` have a lot of NVP values but not have `nvp_dismissedHoldUseExplanation`, also OnyxDB not have it.


https://github.com/Expensify/App/assets/41129870/07c4542d-6bec-4710-b72c-1d3b629f96ef


https://github.com/Expensify/App/assets/41129870/a520aa03-44dd-40a0-b776-0fd0a7ad0694

**Comment by cdOut** [comment]
@robertjchen I've checked myself and the nvp is not present in OnyxDB, there also might be a problem with the API write command, since `WRITE_COMMANDS.SET_NAME_VALUE_PAIR` is only used for this one and `ONYXKEYS.NVP_HAS_SEEN_TRACK_TRAININ`, which also does not appear in the OnyxDB.

**Comment by robertjchen** [comment]
@cdOut That could be it! Similar NVPs used for other settings don't seem to have corresponding API calls. Maybe we can try copying what `NVP_HAS_SEEN_TRACK_TRAININ` has done and see if that works?

File: src/libs/actions/IOU.ts [hunk 2]
```diff
@@ -6677,8 +6678,23 @@ function setMoneyRequestTaxAmount(transactionID: string, taxAmount: number | nul
     Onyx.merge(`${ONYXKEYS.COLLECTION.TRANSACTION_DRAFT}${transactionID}`, {taxAmount});
 }
 
-function setShownHoldUseExplanation() {
-    Onyx.set(ONYXKEYS.NVP_HOLD_USE_EXPLAINED, true);
+function dismissHoldUseExplanation() {
+    const parameters: SetNameValuePairParams = {
+        name: ONYXKEYS.NVP_DISMISSED_HOLD_USE_EXPLANATION,
```
Comment (robertjchen) [inline src/libs/actions/IOU.ts:6683]:
```suggestion
        name: ONYXKEYS.NVP_DISMISSED_HOLD_USE_EXPLANATION.replace('nvp_',''),
```

File: src/libs/actions/IOU.ts [hunk 3]
```diff
@@ -6677,8 +6678,23 @@ function setMoneyRequestTaxAmount(transactionID: string, taxAmount: number | nul
...
+        value: true,
+    };
+
+    const optimisticData: OnyxUpdate[] = [
+        {
+            onyxMethod: Onyx.METHOD.MERGE,
+            key: ONYXKEYS.NVP_DISMISSED_HOLD_USE_EXPLANATION,
```
Comment (robertjchen) [inline src/libs/actions/IOU.ts:6690]:
```suggestion
            key: ONYXKEYS.NVP_DISMISSED_HOLD_USE_EXPLANATION.replace('nvp_',''),
```

**Review by robertjchen — CHANGES_REQUESTED** [review]
@cdOut This tweak seemed to have worked for me, I think the prefix might be throwing things off when setting the NVP.

**Comment by robertjchen** [comment]
Going to explore a backend approach 👀

**Comment by robertjchen** [comment]
Got a backend fix ready 👍

**Comment by cdOut** [comment]
@robertjchen will any adjustments have to be made on the frontend side also?

**Comment by robertjchen** [comment]
No, just the backend 👍

**Review by robertjchen — APPROVED** [review]

**Comment by robertjchen** [comment]
## Reviewer Checklist

- [ ] I have verified the author checklist is complete (all boxes are checked off).
- [ ] I verified the correct issue is linked in the `### Fixed Issues` section above
- [ ] I verified testing steps are clear and they cover the changes made in this PR
    - [ ] I verified the steps for local testing are in the `Tests` section
    - [ ] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [ ] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [ ] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [ ] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I verified tests pass on **all platforms** & I tested again on:
    - [ ] Android: Native
    - [ ] Android: mWeb Chrome
    - [ ] iOS: Native
    - [ ] iOS: mWeb Safari
    - [ ] MacOS: Chrome / Safari
    - [ ] MacOS: Desktop
- [ ] If there are any errors in the console that are unrelated to this PR, I either fixed them (preferred) or linked to where I reported them in Slack
- [ ] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [ ] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [ ] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [ ] I verified that comments were added to code that is not self explanatory
    - [ ] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [ ] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
    - [ ] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [ ] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is either coming verbatim from figma or has been approved by marketing (in order to get marketing approval, ask the Bug Zero team member to add the Waiting for copy label to the issue)
    - [ ] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [ ] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [ ] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [ ] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [ ] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [ ] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [ ] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [ ] If a new component is created I verified that:
    - [ ] A similar component doesn't exist in the codebase
    - [ ] All props are defined accurately and each prop has a `/** comment above it */`
    - [ ] The file is named correctly
    - [ ] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [ ] The only data being stored in the state is data necessary for rendering and nothing else
    - [ ] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [ ] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [ ] All JSX used for rendering exists in the render method
    - [ ] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [ ] If any new file was added I verified that:
    - [ ] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [ ] If a new CSS style is added I verified that:
    - [ ] A similar style doesn't already exist
    - [ ] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG`)
- [ ] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [ ] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [ ] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [ ] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [ ] If the PR modifies the UI (e.g. new buttons, new UI components, changing the padding/spacing/sizing, moving components, etc) or modifies the form input styles:
    - [ ] I verified that all the inputs inside a form are aligned with each other.
    - [ ] I added `Design` label and/or tagged `@Expensify/design` so the design team can review the changes.
- [ ] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [ ] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [ ] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>



</details>

<details>
<summary>Android: mWeb Chrome</summary>



</details>

<details>
<summary>iOS: Native</summary>



</details>

<details>
<summary>iOS: mWeb Safari</summary>



</details>

<details>
<summary>MacOS: Chrome / Safari</summary>



</details>

<details>
<summary>MacOS: Desktop</summary>



</details>

**Comment by robertjchen** [comment]
Completed fresh checklist

**Comment by ahmedGaber93** [comment]
> Corresponding backend changes were merged and I tested this PR locally in concert with those changes to confirm that things were working. To speed things up, I'll merge this as well in parallel.

@robertjchen Thanks for completing reviewer's checklist. But we still have pending change here https://github.com/Expensify/App/pull/40435#discussion_r1628847463 need to fix. I think @cdOut should create a follow up PR to fix this.

**Comment by cdOut** [comment]
I'll create a follow up PR in a sec, I wasn't able to push the changes due to the sudden merge.

**Comment by robertjchen** [comment]
Ah, my apologies! 🙇

**Pull request merged**